services/user-tenant-service/src/test/java/com/sourcefuse/jarc/services/usertenantservice/integration/GroupControllerTests.java

```diff
@@ -0,0 +1,504 @@
+
+    MockCurrentUserSession.setCurrentLoggedInUser(null, null, null);
+
+    Mockito.when(groupRepository.save(any(Group.class))).thenReturn(savedGroup);
+    Mockito
+      .when(userGroupsRepository.save(any(UserGroup.class)))
+      .thenThrow(new ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR));
+
+    mockMvc
+      .perform(
+        MockMvcRequestBuilders
+          .post(basePath)
+          .contentType(MediaType.APPLICATION_JSON)
+          .content(asJsonString(group))
+      )
+      .andExpect(MockMvcResultMatchers.status().isInternalServerError())
+      .andExpect(result ->
+        Assertions.assertTrue(
+          result.getResolvedException() instanceof ResponseStatusException
+        )
+      )
+      .andExpect(result ->
+        Assertions.assertEquals(
+          HttpStatus.INTERNAL_SERVER_ERROR.value(),
+          result.getResponse().getStatus()
+        )
+      );
+    // Verify that the save method was called
+    Mockito.verify(groupRepository).save(any(Group.class));
+    // Verify that the save method was never called
+    Mockito.verify(userGroupsRepository).save(any(UserGroup.class));
+  }
+
+  @Test
+  @DisplayName("Test: case for count success")
+   void testCount_Success() throws Exception {
+    // Mock the behavior of roleRepository.count()
+    Mockito.when(groupRepository.count()).thenReturn(5L); // Mock a count of 5
+
+    // Perform the API call
+    mockMvc
+      .perform(
+        MockMvcRequestBuilders
+          .get(basePath + "/count")
+          .contentType(MediaType.APPLICATION_JSON)
+      )
+      .andExpect(MockMvcResultMatchers.status().isOk())
+      .andExpect(result -> Assertions.assertNotNull(result.getResponse()))
+      .andExpect(MockMvcResultMatchers.jsonPath("$.count").value(5));
+
+    // Verify that roleRepository.count() was called
+    Mockito.verify(groupRepository).count();
+  }
+
+  @Test
+  @DisplayName("Test case should pass for 0 count")
+   void testCount_Empty() throws Exception {
+    // Mock the behavior of roleRepository.count()
+    Mockito.when(groupRepository.count()).thenReturn(0L); // Mock a count of 0
+
+    // Perform the API call
+    mockMvc
+      .perform(
+        MockMvcRequestBuilders
+          .get(basePath + "/count")
+          .contentType(MediaType.APPLICATION_JSON)
+      )
+      .andExpect(MockMvcResultMatchers.status().isOk())
+      .andExpect(result -> Assertions.assertNotNull(result.getResponse()))
+      .andExpect(MockMvcResultMatchers.jsonPath("$.count").value(0));
+
+    // Verify that roleRepository.count() was called
+    Mockito.verify(groupRepository).count();
+  }
+
+  @Test
+  @DisplayName("Test: countGroup with database error")
+   void testCountGroup_DatabaseError() throws Exception {
+    // Mock the behavior of roleRepository.save() to throw an exception
+    Mockito
+      .when(groupRepository.count())
+      .thenThrow(new ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR));
```

Comment:
count method will throw Exception related to postgres will not throw ResponseStatusException on its own